sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -329,7 +332,17 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * @since 1.4.0
    */
   def json(jsonRDD: RDD[String]): DataFrame = {
-    val parsedOptions: JSONOptions = new JSONOptions(extraOptions.toMap)
+    val optionsWithTimeZone = {
```

Comment:
Could we just pass the timezone into `JSONOptions` as a default or resemble `columnNameOfCorruptRecord`  in`JSONOptions` below?

It seems the same logics here duplicated several times and logics to set default values in tests are introduced there which might be not necessary or be able to be removed.